hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java

```diff
@@ -721,9 +723,14 @@ public CacheEventJournal getEventJournal() {
     // creates the given CacheConfig on all members of the cluster synchronously
     private <K, V> void createCacheConfigOnAllMembers(CacheConfig<K, V> cacheConfig) {
         OperationService operationService = nodeEngine.getOperationService();
-        CacheCreateConfigOperation op = new CacheCreateConfigOperation(cacheConfig);
-        InternalCompletableFuture future =
-                operationService.invokeOnTarget(CacheService.SERVICE_NAME, op, nodeEngine.getThisAddress());
-        future.join();
+        CacheCreateConfigOperation op = new CacheCreateConfigOperation(
+                new PreJoinCacheConfig(cacheConfig, false), true, true);
```

Comment:
The `PreJoinCacheConfig`'s serialized form will only include class names as `String`s (when KV types are set) instead of the actual `Class` objects used by `CacheConfig`'s serialized form.

Reply:
Maybe this is also worth to comment, e.g.
```
// wrap the CacheConfig in a PreJoinCacheConfig with KV type class names instead of implementations,
// to prevent de-serialization failures on remote nodes
```